internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -99,6 +99,18 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"api_server_vnet_integration_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"api_server_subnet_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
```

Comment:
We should be able to use the validation function for the subnet Id here
```suggestion
				ValidateFunc: networkValidate.SubnetID,
```